### ISSUE TITLE
Open/Close unchanged file - useability

### ISSUE THREAD

**Issue by zawsx** [marktext/marktext#111]
### Description

If you open an .md and then immediately hit the 'x' icon to close that .md file, you get the 'save' prompt even though no changes have been made - annoying. You then have to hit cancel. That doesn't end it though, as you then need to hit the 'x' icon again to actually close the file. One main use of Mark Text for me is to read .md files as well as edit/create them. This is a UI issue and I guess way down the list.

### Steps to reproduce

1. Open and .md file
2. Hit the 'x' icon to close it - no changes having been made.
3. Notice you encounter the 'save' dialog.  Choose 'cancel' (shouldn't have to cancel as no changes have been made but hey...
4. The .md file remains open. To close it, you need to hit 'x' again. Only then does it disappear.

**Expected behavior:** Open any .md file, do not touch the interface except maybe to scroll down to read the full text, then hit 'X' to close it with no further UI interaction.

**Actual behavior:** Several clicks before you can exit the .md, even though it hasn't been changed.

No big deal, but it's be nice if it could be nice and quick. 

### Versions

- Mark Text: 0.9.10 master branch
- OS: W10 Pro x64

**Comment by fxha** [comment]
I noticed a similar problem. The problem here is that the save command (`Ctrl+S`) saves the file only, but don't set "save status" to `true`. We have to `commit('SET_SAVE_STATUS', true)` after each save.

The reasons for your problem seems to be more or less the same. Somewhere we set save status to `false` after we load a new file.

@Jocs Maybe you can fix this? Or I will do sometime.

**Comment by Jocs** [comment]
@fxha I needs your help 👍

**Comment by zawsx** [comment]
Maybe, when a file is first opened and before any changes have been made, it needs to start with commit('SET_SAVE_STATUS', true) [if SET_SAVE_STATUS means: 'false if needing saving']
Some editors have a neat feature that if you undo an edit that effectively resets a file's contents to the original state, the files' save status is reset to true. The result is, if you open a file, change one thing then  undo that single change, no need to flag the file as needing saving, as it hasn't changed from when you read it in.

**Comment by fxha** [comment]
File state is automatically set to `true` (everything fine) after we load a file. Somewhere else it's set to `false` after loading. I'll have to look at it sometime and the "undo" feature is really cool.

**Comment by fxha** [comment]
I fixed the save problems in PR #117.

**TL;DR:** Let's say the parser formats markdown content that causes the changes and this can't be fixed so easily.

But the core problem is, that we parse markdown and translate it into HTML. If you are not in Source Code mode we will translate HTML into markdown and save it internally. If you modify text we update markdown from HTML also. That means internal markdown content and file content may be different, so modification indicator is displayed. We could ignore the first event, but mehhh... I guess this is a wontfix. Maybe @Jocs can help if he implement undo/redo (in Source Code mode), which is also a little bit buggy.

@Jocs I don't know why, but sometimes Mark Text sets a new line at the end of the file - maybe create a settings option (`newLineAtEndOfFile: true/false`). I think this is the same problem as above.

**Comment by zawsx** [comment]
Thanks for the fix - really appreciated :)
The undo issue I'm highlighting is about setting a flag. The flag starts out unset, converting the input to HTML doesn't set it. Changing anything that needs a save does but if you faithfully undo those changes so as to revert to the unchanged original, it undoes the flag setting at the point where you've brought the file back to it's original contents - hence no need to save it again.
This is such a tiny issue, could usefully be ignored - it's not a bug.
